src/components/steps/ingredients/ingredients.tsx

```diff
@@ -1,72 +1,135 @@
 import {
   Button,
-  Chip,
   IconButton,
   MenuItem,
-  Stack,
   TextField,
   Typography,
 } from '@mui/material';
 
 import ArrowRightAltIcon from '@mui/icons-material/ArrowRightAlt';
 import CloseIcon from '@mui/icons-material/Close';
 
+import { Instruction, Section, Component } from 'utils/api/api.types';
+
 import './ingredients.scss';
 
+import { StepsProps } from '../nameAndDescription/nameAndDescription';
+
 type FormValues = {
   ingredient: '';
-  instruction: '';
+  instructionInput: '';
 };
 
 const getServingOptions = () => {
   let n = 1;
-  const numbers: number[] = [];
+  const numbers: string[] = [];
   while (n <= 30) {
-    numbers.push(n);
+    numbers.push(n.toString());
     n++;
   }
+
   return numbers;
 };
 const servingsOptions = getServingOptions();
 
-const Ingredients = () => {
-  const [ingredientsList, setIngredientsList] = useState<string[]>([]);
-  const [instructionsList, setInstructionsList] = useState<string[]>([]);
+type Instructions = Instruction[];
+type Components = Component[];
+
+const Ingredients: React.FC<StepsProps> = ({ submitStep, setSubmitStep }) => {
+  const [components, setComponents] = useState<Components>([]);
+  const [instructions, setInstructions] = useState<Instructions>([]);
+  const [yields, setYields] = useState<string>('Servings: 1');
+
+  const { recipeToUpload, setDisplayMessage, setRecipeToUpload } =
+    useContext(StorageContext);
+
+  useEffect(() => {
+    if (recipeToUpload) {
+      if (recipeToUpload.instructions) {
+        setInstructions(recipeToUpload.instructions);
+      }
+      if (recipeToUpload.sections) {
+        setComponents(recipeToUpload.sections[0].components);
+      }
+      if (recipeToUpload?.yields) {
+        setYields(recipeToUpload?.yields);
+      }
+    }
+  }, []);
+
+  useEffect(() => {
+    if (submitStep.initialized) {
+      try {
+        if (!instructions || !components || !yields) {
+          setSubmitStep({ ...submitStep, initialized: false });
+          throw new Error('All fields are required.');
+        }
+
+        const sections = [{ components }];
+        setRecipeToUpload({
+          ...recipeToUpload,
+          sections,
+          instructions,
+          yields,
+        });
+
+        setSubmitStep({ initialized: false, submitted: true });
+      } catch (error) {
+        setDisplayMessage((error as Error).message);
+      }
+    }
+  }, [submitStep.initialized]);
 
   const { handleSubmit, register, reset } = useForm<FormValues>({
-    defaultValues: { ingredient: '', instruction: '' },
+    defaultValues: { ingredient: '', instructionInput: '' },
   });
 
   const onSubmit = (data: FormValues) => {
-    const { ingredient, instruction } = data;
+    const { ingredient, instructionInput } = data;
 
     if (ingredient) {
-      setIngredientsList([...ingredientsList, ingredient]);
+      setComponents([
+        ...components,
+        {
+          id: (components[components.length - 1]?.id as number) + 1 || 0,
```

Comment:
This doesn't seem right. You should avoid doing type castings. Check if wouldn't be possible to declare the id field on the Component type as a number